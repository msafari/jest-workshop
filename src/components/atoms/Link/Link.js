import React from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'

const propTypes = {
  linkName: PropTypes.string.isRequired
}

const StyledLink = styled.a`
  color: orange;
`

const Link = ({
  linkName
}) => {
  return (
    <StyledLink>
      Click here to open {linkName}
    </StyledLink>
  )
}

Link.propTypes = propTypes

export default Link
