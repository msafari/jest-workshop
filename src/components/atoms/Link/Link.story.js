import React from 'react'
import { storiesOf } from '@storybook/react'
import Link from '.'

const story = storiesOf('atoms/Link', module)
  .add('without props', () => {
    return (<Link/>)
  })
  .add('with required props', () => {
    return (<Link linkName='Priceline'/>)
  })

export default story
