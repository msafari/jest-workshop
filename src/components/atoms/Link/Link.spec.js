import React from 'react'
import renderer from 'react-test-renderer'
import Link from '.'

test('Link component renders correctly', () => {
  const link = renderer.create(
    <Link linkName='Priceline'/>
  ).toJSON()
  expect(link).toMatchSnapshot()
})
