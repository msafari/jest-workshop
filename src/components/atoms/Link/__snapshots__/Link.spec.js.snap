// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link component renders correctly 1`] = `
<a
  className="sc-bdVaJa iWzBfR"
>
  Click here to open 
  Priceline
</a>
`;
